WordPress/Classes/ViewRelated/Reader/ReaderPostHeaderView.m

```diff
@@ -22,185 +24,184 @@ - (instancetype)initWithFrame:(CGRect)frame
-                                                                   views:views]];
-    [self addConstraint:[NSLayoutConstraint constraintWithItem:_disclosureButton
-                                                     attribute:NSLayoutAttributeCenterY
-                                                     relatedBy:NSLayoutRelationEqual
-                                                        toItem:_avatarImageView
-                                                     attribute:NSLayoutAttributeCenterY
-                                                    multiplier:1.0f
-                                                      constant:0.0f]];
-    [super setNeedsUpdateConstraints];
-}
-
-
-#pragma mark - Subview factories
-
-- (UILabel *)newLabelForTitle
-{
-    UILabel *label = [[UILabel alloc] initWithFrame:CGRectZero];
+    UILabel *label = [[UILabel alloc] init];
     label.translatesAutoresizingMaskIntoConstraints = NO;
     label.numberOfLines = 1;
     label.backgroundColor = [UIColor whiteColor];
     label.opaque = YES;
-    label.textColor = [WPStyleGuide littleEddieGrey];
+    label.textColor = [WPStyleGuide allTAllShadeGrey];
     label.font = [WPStyleGuide subtitleFont];
     label.adjustsFontSizeToFitWidth = NO;
 
-    return label;
+    [self.labelsStackView addArrangedSubview:label];
+    self.subtitleLabel = label;
 }
 
-- (UILabel *)newLabelForSubtitle
+- (void)setupTitleLabel
 {
-    UILabel *label = [[UILabel alloc] initWithFrame:CGRectZero];
+    NSAssert(self.labelsStackView != nil, @"labelsStackView was nil");
+
+    UILabel *label = [[UILabel alloc] init];
     label.translatesAutoresizingMaskIntoConstraints = NO;
     label.numberOfLines = 1;
     label.backgroundColor = [UIColor whiteColor];
     label.opaque = YES;
-    label.textColor = [WPStyleGuide allTAllShadeGrey];
+    label.textColor = [WPStyleGuide littleEddieGrey];
     label.font = [WPStyleGuide subtitleFont];
     label.adjustsFontSizeToFitWidth = NO;
 
-    return label;
+    [self.labelsStackView addArrangedSubview:label];
+    self.titleLabel = label;
 }
 
-- (CircularImageView *)newImageViewForAvatar
+- (void)setupDisclosureButton
 {
-    CGRect avatarFrame = CGRectMake(0.0f, 0.0f, PostHeaderViewAvatarSize, PostHeaderViewAvatarSize);
-    CircularImageView *imageView = [[CircularImageView alloc] initWithFrame:avatarFrame];
-    imageView.translatesAutoresizingMaskIntoConstraints = NO;
-    return imageView;
+    NSAssert(self.stackView != nil, @"stackView was nil");
+
+    UIImage *chevronImage = [[UIImage imageNamed:@"disclosure-chevron"] imageFlippedForRightToLeftLayoutDirection];
+    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+    button.translatesAutoresizingMaskIntoConstraints = NO;
+    [button setBackgroundImage:chevronImage forState:UIControlStateNormal];
+
+    NSLayoutConstraint *width = [button.widthAnchor constraintEqualToConstant:chevronImage.size.width];
+    width.priority = 999;
```

Comment:
For this one, since the `disclosureButton` can be set to `hidden`, we need a non-required priority for the width layout, within `UIStackView`. And, `999` works better than `UILayoutPriorityHigh` because we don't want to actually discourage the width constraint from doin it's thang.